### PR TITLE
adios2: fix unresolved symbols in 2.6.0 when built with gcc10

### PR DESCRIPTION
**Pull request by germasch**
This fixes unresolved symbols when building adios2@2.6.0 with gcc10.

A related question: Should PRs like this go against `develop` or `releases/v0.16`? It looks like support for newer adios2 versions (up to 2.7.1) has been added to `develop` back in February, but it's still not in spack 0.16.2.

### CODE REVIEW

**Comment by chuckatkins** [comment]
I've marked this as "don't merge yet" until it's merged upstream.  At that time this should use whatever the final upstream patch is.

**Comment by germasch** [comment]
@chuckatkins Could you do something to help move this along then? Or should I do anything?

**Review by chuckatkins — APPROVED** [review]

**Pull request merged**
